### PR TITLE
feat: add mac release zip file and release info yml

### PR DESCRIPTION
**Pull request by m1911star**
image

https://github.com/toeverything/AFFiNE/actions/runs/4829332390

purpose:
we will use electron-updater as a base module to implement auto updater with channel(canary/beta/stable). But we use electron-forge as builder. So we need to generate assets with script

ref: 
- https://www.electron.build/auto-update.html
- https://github.com/electron-userland/electron-builder/blob/master/packages/electron-updater/README.md

### CODE REVIEW

File: apps/electron/package.json [hunk 1]
```diff
@@ -44,6 +45,7 @@
     "electron-window-state": "^5.0.3",
     "esbuild": "^0.17.18",
     "fs-extra": "^11.1.1",
+    "js-yaml": "^4.1.0",
```
Comment (pengx17) [inline apps/electron/package.json:48]: I think you did not use this library in generate-yml. Maybe remove it or generate JSON and convert it to yml using this lib instead?

File: apps/electron/package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
     "make-macos-x64": "electron-forge make --platform=darwin --arch=x64",
     "make-windows-x64": "electron-forge make --platform=win32 --arch=x64",
     "make-linux-x64": "electron-forge make --platform=linux --arch=x64",
+    "generate-release-yml": "zx scripts/generate-yml.mjs",
```
Comment (pengx17) [inline apps/electron/package.json:18]: It seems this script does not exist and `generate-yml.js` does not use zx. You can remove it.

File: apps/electron/package.json (hunk 1 above)
Comment (m1911star) [reply apps/electron/package.json]: will use fs.writeFileSync to write a yml string

File: apps/electron/package.json (hunk 2 above)
Comment (m1911star) [reply apps/electron/package.json:18]: removed

File: .github/workflows/release-desktop-app.yml [hunk 3]
```diff
@@ -169,7 +175,17 @@ jobs:
         with:
           name: affine-linux-x64-builds
           path: ./
-
+      - name: Download Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: release-yml-build-script
+          path: ./
+      - uses: actions/setup-node@v3
```
Comment (pengx17) [inline .github/workflows/release-desktop-app.yml:183]: may use ./.github/actions/setup-node instead

File: .github/workflows/release-desktop-app.yml [hunk 4]
```diff
@@ -169,7 +175,17 @@ jobs:
...
+        with:
+          node-version: 18
+      - name: Generate Release yml
+        run: |
+          BUILD_TYPE=${{ env.BUILD_TYPE }} RELEASE_VERSION=${{ github.event.inputs.version }} node generate-yml.js
```
Comment (pengx17) [inline .github/workflows/release-desktop-app.yml:188]: I think you do not need to set BUILD_TYPE here, which is already available in the workflow globally

File: .github/workflows/release-desktop-app.yml (hunk 3 above)
Comment (m1911star) [reply .github/workflows/release-desktop-app.yml:183]: in the final release step we only need a simple nodejs env, dont need to checkout repo or restore cache, so maybe we dont need to use the exist action

File: .github/workflows/release-desktop-app.yml (hunk 4 above)
Comment (m1911star) [reply .github/workflows/release-desktop-app.yml:188]: not sure whether can access build type in generate-yml.js

Comment (m1911star) [reply .github/workflows/release-desktop-app.yml]: checked & removed

**Review by pengx17 — APPROVED** [review]

**Pull request merged**